.codegen/impl.java.tmpl

```diff
@@ -26,18 +28,36 @@ class {{.PascalName}}Impl implements {{.PascalName}}Service {
     String path = {{if .PathParts -}}
       String.format("{{range  .PathParts}}{{.Prefix}}{{if or .Field .IsAccountId}}%s{{end}}{{ end }}"{{ range .PathParts }}{{if .Field}}, request.get{{.Field.PascalName}}(){{ else if .IsAccountId }}, apiClient.configuredAccountID(){{end}}{{ end }})
     {{- else}}"{{.Path}}"{{end}};
-    {{if .Response -}}
-      {{- if .Response.ArrayValue -}} return apiClient.getCollection(path, null, {{template "type" .Response.ArrayValue}}.class);
-      {{- else if .Response.MapValue -}} return apiClient.getStringMap(path, request);
-      {{- else -}} return apiClient.{{.Verb}}(path{{if .Request}}, request{{end}}{{if .Response -}}, {{if .Response.ArrayValue }}Collection
-                        {{- else if .Response.MapValue }}Map
-                        {{- else}}{{template "type" .Response}}
-                        {{- end -}}{{else}}, Void{{end}}.class);{{end}}
-    {{- else -}}apiClient.{{.Verb}}(path{{if .Request}}, request{{end}}{{if .Response -}}, {{if .Response.ArrayValue }}Collection
-      {{- else if .Response.MapValue }}Map
-      {{- else}}{{template "type" .Response}}
-      {{- end -}}{{else}}, Void{{end}}.class);
-    {{end}}
+    {{ template "headers" . -}}
+    {{ if not .Response -}}
+    {{ template "api-call" . }}
+    {{- else if .Response.ArrayValue -}} return apiClient.getCollection(path, null, {{template "type" .Response.ArrayValue}}.class, headers);
+    {{- else if .Response.MapValue -}} return apiClient.getStringMap(path, {{ template "request-param" .}}, headers);
+    {{- else if .IsResponseByteStream -}}
+    InputStream response = {{ template "api-call" . }}
+    return new {{ .Response.PascalName }}().setContents(response);
+    {{- else }}return {{ template "api-call" . }}
+    {{- end}}
   }
   {{end}}
-}
\ No newline at end of file
+}
+
+{{ define "api-call" }}
+apiClient.{{.Verb}}(path
+  {{- if .Request}}, {{ template "request-param" .}}{{end}}
+  , {{ if not .Response -}}Void
+  {{- else if .IsResponseByteStream}}InputStream
+  {{- else}}{{template "type" .Response}}{{- end -}}.class
+  , headers);
+{{- end }}
+
+{{ define "headers" -}}
+    Map<String, String> headers = new HashMap<>();
```

Comment:
this results in the unnecessary diff and headers variable for literally every single method. surround it with `if` and make sure there's no diff in the unaffected methods
```suggestion
    {{if .FixedRequestHeaders}}
    ... what you have here
    {{end}}
```